File: pi/zpl.py
def generate_zpl(url: str) -> str:
    return (
        "^XA"
        "^FO50,50"
        f"^BQN,2,5"
        "^FH"
        f"^FDQA,{url}^FS"
        "^FO50,250"
        "^A0N,28,28"
        f"^FD{url}^FS"
        "^XZ"
    )

File: worker/src/zpl.ts
const LABEL_WIDTH_DOTS = 609;
const QR_SIZE = 8;

export function generateZpl(url: string): string {
  const xPos = Math.floor((LABEL_WIDTH_DOTS - QR_SIZE * 25) / 2);
  return `^XA\n^FO${xPos},50\n^BQN,2,${QR_SIZE}\n^FDQA,${url}^FS\n^XZ`;
}
